src/controllers/action-queue/runtime.js

```diff
@@ -46,6 +46,39 @@ async function checkActionEffect(account: EdgeAccount, effect: ActionEffect): Pr
       })
       return (await Promise.all(promises)).every(yes => yes)
     }
+    case 'address-balance': {
+      // TODO: Use effect.address when we can check address balances
+      const { aboveAmount, belowAmount, tokenId, walletId } = effect
+      const wallet = account.currencyWallets[walletId]
+      const currencyCode = getCurrencyCode(wallet, tokenId)
+      const walletBalance = wallet.balances[currencyCode] ?? '0'
+
+      return (aboveAmount != null && gt(walletBalance, aboveAmount)) || (belowAmount != null && lt(walletBalance, belowAmount))
+    }
+    case 'tx-confs': {
+      const { txId, walletId, confirmations } = effect
+      const wallet = account.currencyWallets[walletId]
+
+      // Get transaction
+      const txs = await wallet.getTransactions({
+        // TODO: Add a parameter to limit to one transaction in result
+        searchString: txId
+      })
+      if (txs.length === 0) throw new Error(`Cannot find transaction with txid '${txId}' in '${wallet.id}'`)
```

Comment:
Instead of failing, shouldn't we conclude that the confirmations are 0? We might be racing with engines that haven't seen it yet.

Reply:
Agree